transformer/scenario.py

```diff
@@ -0,0 +1,277 @@
+        ]
+        if dangling_weight_files:
+            hint = ", ".join(str(f) for f in dangling_weight_files)
+            logging.error(
+                "%s contains weight files that don't correspond to any scenarios: %s",
+                path,
+                hint,
+            )
+            logging.info(
+                "For any value of X, if there exists a weight file X.weight, "
+                "there must exist either an X.har file or an X scenario subdirectory."
+            )
+            raise DanglingWeightError(path, "contains dangling weight files")
+
+    @classmethod
+    def from_har_file(
+        cls, path: Path, plugins: Sequence[Plugin], short_name: bool
+    ) -> "Scenario":
+        """
+        Creates a Scenario given a HAR file.
+
+        :raise SkippableScenarioError: if path is unreadable or not a HAR file
+        """
+        try:
+            with path.open() as file:
+                har = json.load(file)
+            requests = Request.all_from_har(har)
+            tasks = list(Task.from_requests(requests))
+
+            for plugin in plugins:
+                tasks = plugin(tasks)
+
+            return Scenario(
+                name=to_identifier(
+                    path.with_suffix("").name if short_name else str(path)
+                ),
+                children=tuple(tasks),
+                origin=path,
+                weight=cls.weight_from_path(path),
+            )
+        except (OSError, json.JSONDecodeError, UnicodeDecodeError) as err:
+            raise SkippableScenarioError(path, err)
+
+    @classmethod
+    def weight_from_path(cls, path: Path) -> int:
+        """
+        Reads the weight file corresponding to path, or returns a default weight
+        if the weight file doesn't exist.
+
+        :param path: represents either a HAR file or a scenario directory
+        :raise WeightValueError: if the weight file exists but its contents cannot be
+            interpreted as a weight
+        """
+        weight_path = path.with_suffix(WEIGHT_FILE_SUFFIX)
+
+        try:
+            weight = weight_path.read_text().strip()
+        except OSError as err:
+            logging.info(
+                f"No {weight_path} provided for {path}: "
+                f"assigning default weight {DEFAULT_WEIGHT} ({err})"
+            )
+            return DEFAULT_WEIGHT
+
+        if not weight.isdecimal() or int(weight) == 0:
+            logging.error(
+                f"invalid weight file %s: weights must be positive integers, got %r",
+                weight_path,
+                weight,
+            )
+            raise WeightValueError(path, weight)
+
+        return int(weight)
+
+    @property
+    def global_code_blocks(self) -> Mapping[str, Sequence[str]]:
+        # TODO: Replace me with a plugin framework that accesses the full tree.
+        #   See https://github.bus.zalan.do/TIP/docs/issues/395.
```

Comment:
Do we have an external link for this ticket? Or how should we handle this do you think?

Reply:
I will update these links as soon as this PR is merged, but we need these files (`scenario.py` and `task.py`) to create the corresponding issue, which we can then link to in the files :upside_down_face: